template/errors.go

```diff
@@ -0,0 +1,52 @@
+package template
+
+import (
+	"errors"
+	"fmt"
+
+	ast "go.expect.digital/mf2/parse"
+)
+
+// MessageFormat2 Errors as defined in the specification.
+//
+// https://github.com/unicode-org/message-format-wg/blob/122e64c2482b54b6eff4563120915e0f86de8e4d/spec/errors.md
+var (
+	ErrSyntax                = errors.New("syntax error")
+	ErrUnresolvedVariable    = errors.New("unresolved variable")
+	ErrUnknownFunction       = errors.New("unknown function reference")
+	ErrDuplicateOptionName   = errors.New("duplicate option name")
+	ErrUnsupportedExpression = errors.New("unsupported expression")
+	ErrFormatting            = errors.New("formatting error")
+)
+
+func syntaxErr(err error) error {
+	return errors.Join(ErrSyntax, err)
```

Comment:
We MUST support golang 1.18. `errors.Join()` was added in v1.20 https://pkg.go.dev/errors#Join

Reply:
i see, looking for altenative